cranelift/fuzzgen/src/lib.rs

```diff
@@ -218,14 +229,76 @@ where
         self.run_func_passes(func)
     }
 
+    /// Generate a random set of cranelift flags.
+    /// Only semantics preserving flags are considered
+    fn generate_flags(&mut self) -> Result<Flags> {
+        let mut builder = settings::builder();
+
+        let opt = self.u.choose(OptLevel::all())?;
+        builder.set("opt_level", &format!("{}", opt)[..])?;
+
+        // Boolean flags
+        // TODO: probestack is semantics preserving, but only works inline and on x64
+        // TODO: enable_pinned_reg does not work with our current trampolines. See: #4376
+        // TODO: is_pic has issues:
+        //   x86: https://github.com/bytecodealliance/wasmtime/issues/5005
+        //   aarch64: https://github.com/bytecodealliance/wasmtime/issues/2735
+        let bool_settings = [
+            "enable_alias_analysis",
+            "enable_simd",
+            "enable_float",
+            "enable_atomics",
+            "enable_safepoints",
+            "unwind_info",
+            "preserve_frame_pointers",
+            "machine_code_cfg_info",
+            "enable_jump_tables",
+            "enable_heap_access_spectre_mitigation",
+            "enable_table_access_spectre_mitigation",
+            "enable_incremental_compilation_cache_checks",
+            "regalloc_checker",
+            "enable_llvm_abi_extensions",
+            "use_egraphs",
```

Comment:
Some of these can probably be excluded, based on knowledge of what they do and likely payoff from fuzzing time:

* `machine_code_cfg_info` generates additional metadata for the embedder but this doesn't feed back into compilation anywhere; we can leave it on unconditionally to make sure the generation doesn't panic
* `enable_simd`, `enable_float`, `enable_atomics` will cause panics in compilation if an input uses one of those features and it isn't enabled, but don't change compilation otherwise; so we can probably turn them on unconditionally (we're not, e.g., trying to test that a SIMD input causes panics with the feature off)

Reply:
Agreed, let me know if any other flags come to mind!

> `enable_simd`, `enable_float`, `enable_atomics` will cause panics in compilation if an input uses one of those features and it isn't enabled,

Enabling them unconditionally sounds good.

But at least for `enable_floats` it hasn't panicked yet and we almost always insert floats for larger programs, so we might be silently allowing them anyway. (Haven't tested this yet)